### ISSUE TITLE
[docs] only build and deploy master

### ISSUE THREAD

**Issue by ydcjeff** [pytorch/ignite#1898]
## 📚 Documentation
We are currently building the docs from all tags / releases using master template and configs. And there are a bunch of warnings for past releases. This can lead to docs building error if we are going to bump sphinx or use BC breaking new features from sphinx. So, we should only build and deploy from master and a tag for a release. Since we only build for master, we can enable `-W` option which treats warnings as an error and thus docs build will fail, making sure the deployed docs is free of warnings.

Question is how to manage versioned docs ?
We could setup docs build like domain repositories but we'll see if there are any other options

**Comment by vfdev-5** [comment]
@ydcjeff yes, we are building all versions and also do no-history upload to gh-pages (to avoid repository size explosion). I'm not sure about how to build only master and the latest tag, keeping previous tags without rebuilding and do no-history upload...

**Comment by ydcjeff** [comment]
To keep building master and latest tag and avoid rebuilding for old docs, why don't we use https://readthedocs.org ?

**Comment by vfdev-5** [comment]
Readthedoc is a service and I think it should be paid if using our own server.

**Comment by ydcjeff** [comment]
But it is free for open source, right?
